### PR TITLE
CORTX-25630 : Kubernetes cluster support on bare-metal

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
# Problem Statement
- Kubernetes cluster creation was failing on bare-metal due to calico

# Design
-  Fixed calico pod issue by adding `IP_AUTO_DETECTION` method on interface for bare-metals.
# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
   https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-kubernetes-cluster/3319/
   https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-kubernetes-cluster/3320/
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: solutions/kubernetes/cluster-functions.sh [hunk 1]
```diff
@@ -283,8 +283,12 @@ function setup_primary_node(){
             CALICO_PLUGIN_MAJOR_VERSION=$(echo $CALICO_PLUGIN_VERSION | awk -F[.] '{print $1"."$2}')
             curl https://docs.projectcalico.org/archive/$CALICO_PLUGIN_MAJOR_VERSION/manifests/calico.yaml -o calico-$CALICO_PLUGIN_VERSION.yaml || throw $Exception    
         fi
-        # Setup IP_AUTODETECTION_METHOD for determining calico network.
-        # sed -i '/# Auto-detect the BGP IP address./i \            - name: IP_AUTODETECTION_METHOD\n              value: "interface=eth-0"' calico-$CALICO_PLUGIN_VERSION.yaml
+
+        IS_VM=$(systemd-detect-virt -v)
+        if [ "$IS_VM" == "none" ]; then
+            # Setup IP_AUTODETECTION_METHOD for determining calico network.
+            sed -i '/# Auto-detect the BGP IP address./i \            - name: IP_AUTODETECTION_METHOD\n              value: "interface=eno1"' calico-$CALICO_PLUGIN_VERSION.yaml
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/cluster-functions.sh:290]: Can we avoid using hardcoded `eno1`

Comment (gauravchaudhari02) [reply solutions/kubernetes/cluster-functions.sh:290]: @shailesh-vaidya, if not `eno1` hardcoded, we need to get interface name using ip which starts with 10. is that fine?

File: jenkins/custom-ci/generic/custom-ci.groovy [hunk 2]
```diff
@@ -439,7 +439,7 @@ pipeline {
                 script { build_stage = env.STAGE_NAME }
                 script {
                     try {
-                        def build_cortx_all_image = build job: 'cortx-all-docker-image', wait: true,
+                        def build_cortx_all_image = build job: 'cortx-image', wait: true,
```
Comment (shailesh-vaidya) [inline jenkins/custom-ci/generic/custom-ci.groovy:442]: I guess this change is not intended for CORTX-25630 
Also, we need to trigger - https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/cortx-all-docker-image/

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/custom-ci/generic/custom-ci.groovy (hunk 2 above)
Comment (gauravchaudhari02) [reply jenkins/custom-ci/generic/custom-ci.groovy]: @shailesh-vaidya, Yes. removed it.

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**